### Import & Read CSV

In [1]:
import pandas as pd 

In [2]:
game_df = pd.read_csv('Resources/purchase_data.csv')

In [3]:
game_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

### Player Count 

In [4]:
player_count = len(game_df['SN'].value_counts())
Total_Players = pd.DataFrame({"Total Players": player_count}, index=[0])

Total_Players

Total Players
0            576

### Purchasing Analysis

In [5]:
# number on unique items 
unique_items = len(game_df['Item ID'].unique())

# Average sale price 
avg_price = game_df['Price'].mean()
avg_price = round(avg_price, 2)

#Total number of purchases 
purchase_count = game_df['Purchase ID'].count()

#Total Revenue 
total_revenue = game_df['Price'].sum()

In [6]:
#create summary dataframe
Summary_Stats = pd.DataFrame({"Number of Unique Items": [unique_items],
                           "Average Sales Price": [avg_price],
                           "Total Purchase Count": [purchase_count],
                           "Total Revenue (US Dollars)": [total_revenue]})
Summary_Stats.head()

Number of Unique Items  Average Sales Price  Total Purchase Count  \
0                     179                 3.05                   780   

   Total Revenue (US Dollars)  
0                     2379.77

### Player Demographics (Gender)

In [7]:
gender = game_df.loc[:, ['SN', 'Gender']]
gender = gender.drop_duplicates ()
gender.shape

(576, 2)

In [8]:
gender.head()

SN Gender
0        Lisim78   Male
1    Lisovynya38   Male
2     Ithergue48   Male
3  Chamassasya86   Male
4      Iskosia90   Male

In [9]:
#Gender Count
gender_count = gender['Gender'].value_counts()
gender_count

#Total Gender
total_gender = len(gender)
total_gender

576

In [10]:
# Gender Precentage for Total Players
percentage = gender_count / total_gender
percentage = round(percentage *100, 2)
percentage

Male                     84.03
Female                   14.06
Other / Non-Disclosed     1.91
Name: Gender, dtype: float64

In [11]:
# Create Gender Summary Table
Gender_Summary = pd.DataFrame({"Count" : gender_count,
                               "Percentage" : percentage})


Gender_Summary

Count  Percentage
Male                     484       84.03
Female                    81       14.06
Other / Non-Disclosed     11        1.91

### Purchasing Analysis (Gender)

In [12]:
purchase_count=game_df.groupby(["Gender"]).count()["Price"]
avg_purchase_price=game_df.groupby(["Gender"]).mean()["Price"]
total_purchase_value=game_df.groupby(["Gender"]).sum()["Price"]

g_purchase_total=total_purchase_value/gender_count

summary_purchasing_analysis=pd.DataFrame({"Purchase Count":purchase_count,
                                         "Average Purchase Price":avg_purchase_price,
                                         "Total Purchase Value": total_purchase_value,
                                         "Normalized Total":g_purchase_total})

summary_purchasing_analysis=summary_purchasing_analysis.round(2)
summary_purchasing_analysis

Purchase Count  Average Purchase Price  \
Gender                                                          
Female                            113                    3.20   
Male                              652                    3.02   
Other / Non-Disclosed              15                    3.35   

                       Total Purchase Value  Normalized Total  
Gender                                                         
Female                               361.94              4.47  
Male                                1967.64              4.07  
Other / Non-Disclosed                 50.19              4.56

### Age Demographics 

In [13]:
playerage = game_df.loc[:, ['SN', 'Age']]
playerage = playerage.drop_duplicates()
playerage.head()

SN  Age
0        Lisim78   20
1    Lisovynya38   40
2     Ithergue48   24
3  Chamassasya86   24
4      Iskosia90   23

In [14]:
# Create Bins and Titles
bins = [0, 10, 15, 20, 25, 30, 35, 40, 200]
bin_titles = ['<10', '10-14', '15-19', '20-24', '25-29', '30-34', '35-39', '40+']

playerage["Age Groups"] = pd.cut(playerage["Age"], bins, labels=bin_titles)
player_age_df = playerage.groupby(["Age Groups"])

In [15]:
#Purchase count
age_purchase = playerage["Age Groups"].value_counts()


In [16]:
#total Players
total_players = len(playerage)

# User Age Range Percentages 
agegroup_percentage = round(age_purchase / total_players *100, 2)

In [17]:
#Create Dataframe

Age_Demo_Summary = pd.DataFrame({"Total Count": age_purchase,
                                 "Percentage of Players":agegroup_percentage})


Age_Demo_Summary["Percentage of Players"] = Age_Demo_Summary["Percentage of Players"] 
Age_Demo_Summary = Age_Demo_Summary .reindex(["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"])
Age_Demo_Summary 

Total Count  Percentage of Players
<10             24                   4.17
10-14           41                   7.12
15-19          150                  26.04
20-24          232                  40.28
25-29           59                  10.24
30-34           37                   6.42
35-39           26                   4.51
40+              7                   1.22

### Top Spenders

In [18]:
#Create Data Columns 

#Player Count
player_count=game_df.groupby(["SN"]).count()["Price"]

#Player Average
player_average=game_df.groupby(["SN"]).mean()["Price"]

#Player Total 
player_total=game_df.groupby(["SN"]).sum()["Price"]

In [19]:
user_summary=pd.DataFrame({"Purchase Count": player_count,
                           "Average Purchase": player_average,
                           "Total Purchase Value": player_total})


user_summary.sort_values("Total Purchase Value", ascending=False).round(2).head()

Purchase Count  Average Purchase  Total Purchase Value
SN                                                                 
Lisosia93                 5              3.79                 18.96
Idastidru52               4              3.86                 15.45
Chamjask73                3              4.61                 13.83
Iral74                    4              3.40                 13.62
Iskadarya95               3              4.37                 13.10

### Most Popular Items

In [20]:
#Group Game DF by Item ID
item_id = game_df.groupby(game_df['Item ID'])

#Get Unique Values
unique_items = item_id['Item ID'].unique().str[0]
item_name = item_id['Item Name'].unique().str[0]

#Item Count
item_purchase_count = item_id['Age'].count()
item_price = item_id['Price'].unique().str[0]

#Purchase Total 
item_purchase_total = item_id['Price'].sum()

In [21]:
#Create DataFrame of Most Popular Items

item_purchase_summary = pd.DataFrame({'Item ID':unique_items,
                                      'Item Name':item_name,
                                      'Item Price':item_price,
                                      'Item Count':item_purchase_count,
                                      'Total Purchase':item_purchase_total})

item_purchase_summary = item_purchase_summary.sort_values('Item Count', ascending= False)
item_purchase_summary.index.name = None
item_purchase_summary.head()

Item ID                                     Item Name  Item Price  \
92        92                                  Final Critic        4.88   
178      178  Oathbreaker, Last Hope of the Breaking Storm        4.23   
145      145                          Fiery Glass Crusader        4.58   
132      132                                    Persuasion        3.19   
108      108     Extraction, Quickblade Of Trembling Hands        3.53   

     Item Count  Total Purchase  
92           13           59.99  
178          12           50.76  
145           9           41.22  
132           9           28.99  
108           9           31.77

### Most Profitable Items 

In [22]:
most_profit=item_purchase_summary.sort_values('Total Purchase', ascending=False)
most_profit.index.name = None
most_profit.head()

Item ID                                     Item Name  Item Price  \
92        92                                  Final Critic        4.88   
178      178  Oathbreaker, Last Hope of the Breaking Storm        4.23   
82        82                                       Nirvana        4.90   
145      145                          Fiery Glass Crusader        4.58   
103      103                                Singed Scalpel        4.35   

     Item Count  Total Purchase  
92           13           59.99  
178          12           50.76  
82            9           44.10  
145           9           41.22  
103           8           34.80

### Analysis of Results

1. The player base is largely male, making up 84% of players. Women and other genders only make up 14% and 2% respectivly. 

2. Although males account for the majority of in game purchases, women tend to spend more per purchase than males.

3. Roughly 77% of players are between the ages of 15 and 29, with the largest     portion being between 20-24(40.28%)